modules/claims_api/spec/concerns/claims_api/dependent_claimant_verification_spec.rb

```diff
@@ -76,11 +76,82 @@ class FakeController < ApplicationController
       it 'raises an error' do
         expect do
           subject.validate_dependent_by_participant_id!('', 'any', 'name')
-        end.to raise_error(Common::Exceptions::InvalidFieldValue)
+        end.to raise_error(Common::Exceptions::UnprocessableEntity)
 
         expect do
           subject.validate_dependent_by_participant_id!(nil, 'any', 'name')
-        end.to raise_error(Common::Exceptions::InvalidFieldValue)
+        end.to raise_error(Common::Exceptions::UnprocessableEntity)
+      end
+    end
+  end
+
+  describe '#validate_poa_code_by_participant_id!' do
+    let(:valid_participant_poa_combo) do
+      { ptcpnt_id: '46004', legacy_poa_cd: '002' }
+    end
+
+    let(:another_valid_participant_poa_combo) do
+      { ptcpnt_id: '46028', legacy_poa_cd: '003' }
+    end
+
+    context 'when the participant_id and poa_code are valid' do
+      it 'returns nil and does not raise an error' do
+        VCR.use_cassette('claims_api/bgs/standard_data_web_service/find_poas') do
+          expect do
+            ret = subject.validate_poa_code_by_participant_id!(valid_participant_poa_combo[:ptcpnt_id],
+                                                               valid_participant_poa_combo[:legacy_poa_cd])
+            expect(ret).to eq(nil)
+
+            ret = subject.validate_poa_code_by_participant_id!(another_valid_participant_poa_combo[:ptcpnt_id],
+                                                               another_valid_participant_poa_combo[:legacy_poa_cd])
+            expect(ret).to eq(nil)
+          end.not_to raise_error
+        end
+      end
+    end
+
+    context 'when the participant_id and poa_code are mismatched' do
+      it 'raises an error' do
+        VCR.use_cassette('claims_api/bgs/standard_data_web_service/find_poas') do
+          expect do
+            subject.validate_poa_code_by_participant_id!(valid_participant_poa_combo[:ptcpnt_id],
+                                                         another_valid_participant_poa_combo[:legacy_poa_cd])
+          end.to raise_error(Common::Exceptions::UnprocessableEntity)
+        end
+      end
+    end
```

Comment:
The AC seemed ambiguous to me:

> [When] the POA code requested does NOT have an associated participant id
> Then response returns invalid status

I understood that the POA code must match the associated participant id. But if we should instead check for a valid POA code and the mere presence of the associated participant id, let me know.

Reply:
Update: we only want to confirm the existence of the associated participant ID (we don't need them to match). Fixing.

Reply:
https://github.com/department-of-veterans-affairs/vets-api/pull/18433/commits/bc4bd1d9471c6123b674f4948ef423a13ac294ff